pkg/grpc/device/device.go

```diff
@@ -214,3 +216,33 @@ func PutOwnerCredentialRecord(
 	})
 	return err
 }
+
+var maxCredentialSize = 256 * 1024
+
+// shrinkCredential shrinks a credential object by removing unnecessary responses and options
+// until its within the max credential size
+func shrinkCredential(credential *Credential) {
+	for len(protoutil.NewAny(credential).GetValue()) > maxCredentialSize {
```

Comment:
can there be a situation that this loop will never quit? maybe i.e. limit it to 3 iterations ?

Reply:
No that should not be possible. The loop always breaks if the 3 conditions aren't met, so even if there were nothing left to clear we would not enter an infinite loop. And the 3 conditions all eventually terminate.